tests/pds/resiliency_test.go

```diff
@@ -165,6 +165,131 @@ var _ = Describe("{RebootActiveNodeDuringDeployment}", func() {
+	JustBeforeEach(func() {
+		StartTorpedoTest("RebootNodeDuringAppVersionUpdate", "Reboot node while app version update is going on", pdsLabels, 0)
+	})
+
+	It("Reboot Node While App Version update is going on", func() {
+		var deployments = make(map[PDSDataService]*pds.ModelsDeployment)
+		var generateWorkloads = make(map[string]string)
+
+		Step("Deploy and Validate Data services", func() {
+			for _, ds := range params.DataServiceToTest {
+				deployment, _, _, err = TriggerDeployDataService(ds, params.InfraToTest.Namespace, tenantID, projectID, true)
+				log.FailOnError(err, "Error while deploying data services")
+				err = pdslib.ValidateDataServiceDeployment(deployment, params.InfraToTest.Namespace)
+				log.FailOnError(err, "Error while validating data service deployment")
+				deployments[ds] = deployment
+
+				dataServiceDefaultResourceTemplateID, err = pdslib.GetResourceTemplate(tenantID, ds.Name)
+				log.FailOnError(err, "Error while getting resource template")
+				log.InfoD("dataServiceDefaultResourceTemplateID %v ", dataServiceDefaultResourceTemplateID)
+
+				resourceTemp, storageOp, config, err := pdslib.ValidateDataServiceVolumes(deployment, ds.Name, dataServiceDefaultResourceTemplateID, storageTemplateID, namespace)
+				log.FailOnError(err, "error on ValidateDataServiceVolumes method")
+				ValidateDeployments(resourceTemp, storageOp, config, ds.Replicas, dataServiceVersionBuildMap)
+			}
+		})
+
+		defer func() {
+			for _, newDeployment := range deployments {
+				Step("Delete created deployments")
+				resp, err := pdslib.DeleteDeployment(newDeployment.GetId())
+				log.FailOnError(err, "Error while deleting data services")
+				dash.VerifyFatal(resp.StatusCode, http.StatusAccepted, "validating the status response")
+				err = pdslib.DeletePvandPVCs(*newDeployment.ClusterResourceName, false)
+				log.FailOnError(err, "Error while deleting PV and PVCs")
+			}
+		}()
+
+		Step("Update Data Service Version and reboot node", func() {
+			for ds, deployment := range deployments {
+				// Global Resiliency TC marker
+				pdslib.MarkResiliencyTC(true, true)
+				// Type of failure that this TC needs to cover
+				failuretype := pdslib.TypeOfFailure{
+					Type: ActiveNodeRebootDuringDeployment,
+					Method: func() error {
+						return pdslib.RebootActiveNodeDuringDeployment(params.InfraToTest.Namespace)
+					},
+				}
+				pdslib.DefineFailureType(failuretype)
+
+				log.Infof("Version/Build: %v %v", ds.Version, ds.Image)
+				updatedDeployment, err := pdslib.UpdateDataServiceVerison(deployment.GetDataServiceId(), deployment.GetId(),
+					dataServiceDefaultAppConfigID,
+					int32(ds.Replicas), dataServiceDefaultResourceTemplateID, ds.Image, ds.Version)
+				log.FailOnError(err, "Error while updating data services")
+				log.InfoD("data service %v update triggered", ds.Name)
+
+				err = pdslib.InduceFailureAfterWaitingForCondition(updatedDeployment, namespace, params.ResiliencyTest.CheckTillReplica)
```

Comment:
Issue with this line is , GetPdsSs will always return true immediately in this case without validating that upgrade has triggered or not - as there will be more than 1 pod active here for any multi-node deployment.

Reply:
I have triggered the `UpdateDataServiceVerison ` method just before `InduceFailureAfterWaitingForCondition ` method, and that method handles the issue you mentioned `validating that upgrade has triggered or not`. For me there just be atleast 1 active pods to trigger the reboot.